sw/device/silicon_creator/lib/boot_svc/boot_svc_header.c

```diff
@@ -8,16 +8,89 @@
 
 #include "sw/device/silicon_creator/lib/drivers/hmac.h"
 
-void boot_svc_header_finalize(uint32_t type, uint32_t length,
-                              boot_svc_header_t *header) {
+/**
+ * Computes the digest of a boot services message.
+ *
+ * This function assumes that message payload starts immediately after the
+ * header and is exactly `length - sizeof(boot_svc_header_t)` bytes.
+ *
+ * @param header Header of a boot services message.
+ * @param[out] digest Output buffer for the digest.
+ */
+static void boot_svc_header_digest_compute(const boot_svc_header_t *header,
+                                           hmac_digest_t *digest) {
   enum {
     kDigestRegionOffset = sizeof(header->digest),
   };
   static_assert(offsetof(boot_svc_header_t, digest) == 0,
                 "`digest` must be the first field of `boot_svc_header_t`.");
+
+  hmac_sha256((const char *)header + kDigestRegionOffset,
+              header->length - kDigestRegionOffset, digest);
+}
+
+void boot_svc_header_finalize(uint32_t type, uint32_t length,
+                              boot_svc_header_t *header) {
   header->identifier = kBootSvcIdentifier;
   header->type = type;
   header->length = length;
-  hmac_sha256((const char *)header + kDigestRegionOffset,
-              length - kDigestRegionOffset, &header->digest);
+  boot_svc_header_digest_compute(header, &header->digest);
+}
+
+/**
+ * Shares for producing the `error` value in `boot_svc_header_check()`. First 8
+ * shares are used for checking the digest while the 9th and the 10th shares are
+ * used during the length check. First 10 shares are generated using the
+ * `sparse-fsm-encode` script while the last share is `kErrorOk ^
+ * kErrorBootSvcBadHeader ^ kBootSvcIdentifier ^ kCheckShares[0] ^ ... ^
+ * kCheckShares[9]` so that xor'ing all shares with the initial value of
+ * `error`, i.e. `kErrorBootSvcBadHeader`, and `kBootSvcIdentifier` produces
+ * `kErrorOk`.
+ *
+ * Encoding generated with
+ * $ ./util/design/sparse-fsm-encode.py -d 6 -m 10 -n 32 \
+ *     -s 1606528195 --language=c
```

Comment:
```suggestion
 * ```shell
 *   ./util/design/sparse-fsm-encode.py -d 6 -m 10 -n 32 \
 *     -s 1606528195 --language=c
 * ```
```

Reply:
it looks like this is not something we typically do in `sw/device` but please feel free to create an issue for cleaning up all inline code blocks. Actually, I wasn't even aware that doxygen supported markdown code blocks like this.